policy-engine/src/graph.rs

```diff
@@ -189,4 +240,185 @@ mod tests {
             graph::GraphError::MissingParams(vec!["id".to_string()])
         );
     }
+
+    #[test]
+    fn failed_plugin_execution() -> Result<(), Box<Error>> {
+        use std::str::FromStr;
+
+        let mut rt = common_init();
+        let mandatory_params = vec!["channel".to_string()].into_iter().collect();
+        let state = AppState {
+            mandatory_params,
+            upstream: hyper::Uri::from_str(&mockito::server_url())?,
+            ..Default::default()
+        };
+
+        let http_req = actix_web::test::TestRequest::get()
+            .data(state)
+            .uri(&format!("{}?channel=':'", "http://doesn.really/matter"))
```

Comment:
`doesnt`

But let's use something under `.test` TLD, it's reserved for this exact purpose.